docs/src/markdownDocs/gettingStarted/reactTesting.mdx

```diff
@@ -0,0 +1,145 @@
+
+End-to-end tests usually run in a browser against a live system (in the case of web, using a test runner that automates the browser). Automated E2E tests can take a long time to run because of the different pieces involved and the scope of the tests. You should try to optimize your testing pipeline to run tests in parallel to finish faster.
+
+Functional tests alone will not give you enough test coverage to avoid regressions. Unit tests aim to provide code coverage depth, while functional tests provide coverage over the breadth of user test scenarios.
+
+## How to Write
+
+End-to-end tests rely on the ability to find elements on the screen and interact with them in an automated way. This means that you need a reliable mechanism for selecting elements. Depending on your testing framework, there are different ways to select elements, including:
+
+-   **CSS Selectors**: find elements by using CSS classes or rules (e.g., `.your-class-name`)
+-   **Element Selectors**: find elements inside the elements (e.g., `#\@\@blui-drawerlayout-content > div > header > div > button`)
+-   **xPath Selectors**: find selector within xPath (e.g., `//*[@id="@@blui-drawerlayout-content"]/div/header/div/button`)
+-   **ID Selectors**: find attribute id within elements (e.g., `#login-button`)
+
+> You can read more about different selector strategies on [Selenium Dev](https://www.selenium.dev/documentation/webdriver/elements/locators/) and {/* <!-- markdown-link-check-disable --> */}[software testing help](https://www.softwaretestinghelp.com/using-selenium-xpath-and-other-locators-selenium-tutorial-5/){/* <!-- markdown-link-check-enable --> */}.
+
+Each of these methods has their own advantages and disadvantages. A common pain point in E2E test automation is that modern JavaScript platforms are constantly changing, particularly in the open-source community. Many of these frameworks automatically (and dynamically) generate IDs and classes for components resulting in a constantly moving target for automation tooling. Identifying elements by these values (i.e., using IDs, CSS, or xPath locators _without_ property attributes) makes your tests brittle because they are subject to change any time a new version comes out (or even any time the page is reloaded).
+
+One way to combat this in your own applications is to use dedicated attributes for testing (e.g., `test-id` or `dev-id`). By adding this test-specific attribute to the element, identifying the correct underlying component should be safer. Because this attribute is test-specific and the application logic does not use it, it will be less likely to change throughout the product lifecycle. Combining this approach with other selectors is a common and reliable approach for element identification.
+
+```html
+{/* Component code */}
```

Comment:
<img width="838" alt="Screenshot 2024-11-12 at 3 48 53 PM" src="https://github.com/user-attachments/assets/e00888b6-b1cd-42d9-a3c2-97f98e9331f4">

This comment looks like this.
may need to html comment syntax